### ISSUE TITLE
Radarr deleting movies from library (not disk)

### ISSUE THREAD

**Issue by c0rnflake** [Radarr/Radarr#3128]
This is now the 2nd time in as many months i've ran into this issue.  It could be argued that it's an Unraid issue, but it ain't :)

Anyhow, from what I can work out Unraid has an issue with a disk in my array.  That disk starts to generate read errors.

Once this happens it looks like the disk and the movies contained on the disk disappear from the movies share, /movies.

Now when radarr notices this, i assume during a scheduled task it starts to mark the movies it can no long find the path to as missing, which i do understand why.

BUT i'd prefer if this behaviour didn't happen for obvious reasons those being once Unraid is rebooted the disk errors disappear, the disk is there, radarr can see the paths to the movies is removed from the library.  I then have to revert to a backup library, usually from a week or so ago, then manually go through each movie, editing the path from the old 720p to 1080p release which had been upgraded from when the backup was taken.  I hope that makes sense.

So basically, is there a switch I can use when starting the radarr docker to not mark movies missing in the library if the path doesn't exist anymore?  There is an "Ignore Deleted Movies" switch in the UI, and the hover over stages "Movies deleted from disk are automatically marked unmonitored".  This doesn't seem to be the cash though as they are being marked as missing.

**Comment by TRaSH-** [comment]
Just curious so what would be the issue when it would be marked missing ?
I know in Sonarr it will be also marked as un monitor.
So it wouldn't be re-downloaded.

You say it ain't a a unraid issue but it is,
and now you want Radarr to fix your Unraid issue.
I would check why it happens to your unraid and solve it there first

**Comment by rfgamaral** [comment]
I agree, don't expect the software you use to work flawlessly when you have bigger problems on the system level. Your first problem is your unraid issue, fix that first. You say it's the second time you ran into this issue, do not wait for the problem to become out of your control, find the root cause of your read errors and fix it.

With that said, if you want to work around that issue, I suggest you create a backup task on your system to backup Radarr configuration/database more frequently so that you avoid your restore issues.

**Comment by c0rnflake** [comment]
I've fsck'd the disc outside of the array the first time the issue appeared and no errors where reported.  Ofcourse I'll check it again.

Is the hover over for "Ignore Deleted Movies" wrong?  Why would movies be marked as "missing" when the hover over says "marked as unmonitored"? This is possibly the expected behaviour and the hover over is simply wrong?

**Comment by c0rnflake** [comment]
@galli-leo can you clarify what the "Ignore Deleted Movies" setting actually does, and the actions taken on setting it to Yes and No please.

To me, setting an option to Ignore something to Yes should return a positive response and then not take any actions, ie. setting a movie unmonitored or missing.

**Comment by rfgamaral** [comment]
Let me sum up everything I tried to explain over at Discord:

**Ignore Deleted Movies**: `Yes`

- Radarr scans for disk changes:
  - Is Movie X file missing from the disk when it was available before (ie: did the file got deleted)?
    - YES
      - Stop monitoring Movie X (translation: do not search/download this movie).
      - Set the status as `missing` (translation: this movie does not have a file).
    - NO
      - Do nothing

**Ignore Deleted Movies**: `No`

- Radarr scans for disk changes:
  - Is Movie X file missing from the disk when it was available before (ie: did the file got deleted)?
    - YES
      - Set the status as `missing` (translation: this movie does not have a file).
    - NO
      - Do nothing

What this means is that the "missing" status is **unrelated** to the "Ignore Deleted Movies" option, it will be set to "missing" if the file is missing, whether the option was set to `Yes` or `No`.

No, I'm not 100% sure this is how it works, I'm like 80% or something 😆

**Comment by galli-leo** [comment]
@c0rnflake @rfgamaral This is exactly how the option is supposed to work.

Missing and Unmonitored are two unrelated states.

Maybe this will be better with the upcoming disk scan service changes in proteus, but there is not much I can do here :(
